src/retrieval.rs

```diff
@@ -129,8 +130,108 @@ pub fn fetch_entries<
     }
 }
 
+
+pub fn err(reason: &str) -> WasmError {
+    WasmError::Guest(String::from(reason))
+}
+
+// what is this function doing?
```

Comment:
this is your question? 

I haven't read closely, but assuming that the LinkTag itself (which came from getting the children of a certain day, which would be hours) would contain the info that specifies which hour in particular it is that's 'hanging' here